src/middleware/streamingReceiver.js

```diff
@@ -310,20 +329,37 @@ export async function koaMiddleware (ctx, next) {
   }
 
   if (channel) {
-    collectBody = (channel.matchContentRegex !== null ||
-      channel.matchContentXpath !== null ||
-      channel.matchContentValue !== null ||
-      channel.matchContentJson !== null)
+    collectBody = (
+      channel.matchContentRegex ||
+      channel.matchContentXpath ||
+      channel.matchContentValue ||
+      channel.matchContentJson
+      ) &&
+      ['POST', 'PUT', 'PATCH'].includes(ctx.req.method)
   }
 
-  if (collectBody && ['POST', 'PUT', 'PATCH'].includes(ctx.req.method)) {
-    try {
-      await collectingReceiver(ctx, statusEvents)
-    } catch(err) {
-      logger.error(`collectingReceiver error: ${err}`)
+  if (ctx.isTcpChannel) {
+    if (ctx.tcpChannelHasHttpRoute) {
+      if (collectBody) {
+        try {
+          await collectingReceiver(ctx, statusEvents)
+        } catch(err) {
+          logger.error(`collectingReceiver error: ${err}`)
+        }
+      } else {
+        streamingReceiver(ctx, statusEvents)
+      }
```

Comment:
This functionality seems to be duplicated below, wondering if we cant move this into a re-usable function instead

Reply:
fixed in https://github.com/jembi/openhim-core-js/pull/1063/commits/b4fe0a7d4cb4913c1c81822ec8d5b88ad4c313a3